tests/test_launchable.py

```diff
@@ -0,0 +1,197 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import pytest
+
+from smartsim.database.orchestrator import Orchestrator
+from smartsim.entity.entity import SmartSimEntity
+from smartsim.entity.model import Model
+from smartsim.error.errors import SSUnsupportedError
+from smartsim.launchable import Job, Launchable
+from smartsim.launchable.launchable import SmartSimObject
+from smartsim.launchable.mpmdjob import MPMDJob
+from smartsim.launchable.mpmdpair import MPMDPair
+from smartsim.settings.base import RunSettings
+
+
+def test_smartsimobject_init():
+    ss_object = SmartSimObject()
+    assert isinstance(ss_object, SmartSimObject)
+
+
+def test_launchable_init():
+    launchable = Launchable()
+    assert isinstance(launchable, Launchable)
+
+
+def test_job_init():
+    entity = SmartSimEntity("test_name", None, None)
+    launch_settings = RunSettings("echo", ["spam", "eggs"])
+    job = Job(entity, launch_settings)
+    assert isinstance(job, Job)
+
+
+def test_add_mpmd_pair():
+    """Test the creation of an MPMDJob and then adding an mpmd pair
+    using add_mpmd_pair"""
+    entity = SmartSimEntity("test_name", None, None)
+    launch_settings = RunSettings("echo", ["spam", "eggs"])
+
+    mpmd_job = MPMDJob()
+    mpmd_job.add_mpmd_pair(entity, launch_settings)
+
+    mpmd_pair = MPMDPair(entity, launch_settings)
+
+    assert len(mpmd_job.mpmd_pairs) > 0
```

Comment:
could you instead make sure the length is the exact expected value?

Reply:
todo